src/components/schedule/schedule-time-discover/ScheduleTimeSlot.tsx

```diff
@@ -57,6 +75,51 @@ const ScheduleTimeSlot: FC<ScheduleTimeSlotProps> = ({
     : false
 
   const { isTopElement, isBottomElement } = getMeetingBoundaries(slot, duration)
+
+  // Get current user event from slotData
+  const currentUserEvent = slotData.currentUserEvent
+
+  // Generate calendar URL for the event
+  const eventUrl = useMemo(() => {
+    if (!currentUserEvent) {
+      return null
+    }
+    return generateCalendarEventUrl(
```

Comment:
Why is this not done in the useSlotsWithAvailability where the oicerlapping event is passed ?